toc_distance_calculator.py

```
@@ -0,0 +1,279 @@
+        self.calculation_count += 1
+        
+        try:
+            if self.enable_caching:
+                return self._cached_distance_calculation(lat1, lon1, lat2, lon2)
+            else:
+                return self._haversine_distance(lat1, lon1, lat2, lon2)
+                
+        except Exception as e:
+            raise TOCDistanceCalculatorError(f"Distance calculation failed: {str(e)}") from e
+    
+    @lru_cache(maxsize=10000)
```

Ruff (0.12.2)

Use of `functools.lru_cache` or `functools.cache` on methods can lead to memory leaks

(B019)